src/models/assistant-model.ts

```diff
@@ -115,7 +115,27 @@ export const AssistantModel = types
   }))
   .actions((self) => ({
     addMessageToCODAPNotificationQueue(msg: string) {
-      self.codapNotificationQueue.push(msg);
+      const extracted = extractDataContexts(msg);
+      if (extracted) {
+        self.codapNotificationQueue.push(extracted);
+      }
+    },
```

Comment:
`addMessageToCODAPNotificationQueue` doesn't appear to be used any more.
You could remove it, or update it to be an action that does the dedup and the push, and then call that at line 175 and 213.